src/main/webapp/WEB-INF/view/profile.jsp

```diff
@@ -16,18 +20,29 @@
     <div
       style="width:75%; margin-left:auto; margin-right:auto; margin-top: 50px;">
 
-      <h1>User Profile Pages</h1>
-
       <%
       String currentUser = (String) request.getSession().getAttribute("user");
       String username = (String) request.getAttribute("profilePage");
+      User thisUser = userStore.getUser(username);
 
       if (username.equals("")) { %>
```

Comment:
When does this scenario happen?

Reply:
When you try to retrieve  the username from datastore and it doesn't exist, but I think it returns null instead of "", not sure though!

Reply:
@ppbonds Yeah, @AaronGarcia97 is right so in doGet() when the user doesn't exist/returns null, then I set the "profilePage" parameter to be an empty string. So at the end of doGet() when I forward this request to profile.jsp, it will check if the "profilePage" parameter was set to be an empty string, and if it was, then I know this is because the user doesn't exist. But it should always either return an empty string or a valid username (not null) because I already set it in doGet() :)

here's the code from doGet(): 
// see if user exists
        User user = userStore.getUser(username);

        if (user == null) {
            // if no user exists, set profilePage attribute to be empty string
            request.setAttribute("profilePage", "");
            System.out.println("No user exists.");
        } else {
            // set profilePage attribute to be username
            request.setAttribute("profilePage", username);
        }
        // forward request to profile.jsp
        request.getRequestDispatcher("/WEB-INF/view/profile.jsp").forward(request, response);